Quick/Reporting/XCTestObservationCenter+Reporting.m

```diff
@@ -0,0 +1,182 @@
+    if ([World sharedWorld].reporter) {
+        if (![self qck_isSuspended]) {
+            NSString *testName = [[run test] name];
+            NSDictionary *components = [self qck_getComponentsFromTestName:testName];
+            NSString *testClass = components[@"testClass"];
+            NSString *method = components[@"method"];
+            
+            NSNumber *lineNumber = [NSNumber numberWithUnsignedInteger:line];
+            
+            [[[XCTestDriver sharedTestDriver] IDEProxy] _XCT_testMethod:method ofClass:testClass didMeasureValues:values forPerformanceMetricID:performanceMetricID name:name withUnits:unitsOfMeasurement baselineName:baselineName baselineAverage:baselineAverage maxPercentRegression:maxPercentRegression maxPercentRelativeStandardDeviation:maxPercentRelativeStandardDeviation file:file line:lineNumber];
+            
+            // TODO:
+            // [[TestObservationCenter sharedObservationCenter] testCaseDidMeasureValues...
+        }
+    }
+    else {
+        [self qck_testCase:run didMeasureValues:values forPerformanceMetricID:performanceMetricID name:name unitsOfMeasurement:unitsOfMeasurement baselineName:baselineName baselineAverage:baselineAverage maxPercentRegression:maxPercentRegression maxPercentRelativeStandardDeviation:maxPercentRelativeStandardDeviation maxRegression:maxPercentRelativeStandardDeviation maxStandardDeviation:maxStandardDeviation file:file line:line];
+    }
+}
+
+
+#pragma mark - Helpers
+
++ (void)qck_swizzleInstanceMethod:(SEL)originalSelector withMethod:(SEL)swizzledSelector {
+    Class class = [self class];
+    
+    Method originalMethod = class_getInstanceMethod(class, originalSelector);
+    Method swizzledMethod = class_getInstanceMethod(class, swizzledSelector);
+    
+    BOOL didAddMethod = class_addMethod(class, originalSelector, method_getImplementation(swizzledMethod), method_getTypeEncoding(swizzledMethod));
+    
+    if (didAddMethod) {
+        class_replaceMethod(class, swizzledSelector, method_getImplementation(originalMethod), method_getTypeEncoding(originalMethod));
+    } else {
+        method_exchangeImplementations(originalMethod, swizzledMethod);
+    }
+}
+
+- (NSDictionary *)qck_getComponentsFromTestName:(NSString *)testName {
+    NSUInteger location = 2;
+    NSUInteger length = [testName length] - location - 1;
+    NSRange range = NSMakeRange(location, length);
+    NSString *substring = [testName substringWithRange:range];
+    NSArray *components = [substring componentsSeparatedByString:@" "];
+    
+    NSString *testClass = components[0];
+    NSString *method = [NSString stringWithFormat:@"%@()", components[1]];
+    
+    return @{ @"testClass": testClass, @"method": method };
+}
+
+- (BOOL)qck_isSuspended {
+    return (BOOL)[self valueForKey:@"_isSuspended"];
```

Comment:
This always returns `YES` now, even on projects other than Quick that aren't calling private methods to suspend observation..